src/libraries/Common/src/System/Security/Cryptography/RsaPaddingProcessor.cs

```diff
@@ -30,6 +28,8 @@ internal static int BytesRequiredForBitCount(int keySizeInBits)
             return (int)(((uint)keySizeInBits + 7) / 8);
         }
 
+        private static ReadOnlySpan<byte> EightZeros => new byte[8]; // rely on C# compiler optimization to eliminate allocation
```

Comment:
In asm, [it seems](https://sharplab.io/#v2:C4LghgzgtgNAJiA1AHwAICYCMBYAUBgZjwG88ACCsgBwCcBLANzGAFMzVMA2MgJRbDgB5AHYAbAJ4BlKmGEAeAEbjWAPjIKArgGMA1i2ABBMgF41wlgHd1ylgG0AHAF0A3GQD0bsjRYSyAe2EyAGEAYjItPygqOlEWGn8qYDooOgAvZjoAsmA/Mh9kumFmNjBRUT8tDIDySlpGYvYua1ZbR3VtPWAAIRMzS2a7J2c8AF88IA) to regress. Is there any potential upside of using ROS in this case?

Reply:
@am11 `get_EightZeros` will always be inlined, you just asked sharplab to disasm it without usage

Reply:
Also, that link has => instead of = in bucketB.

Reply:
Here's a local test...
```C#
using System;
using System.Runtime.CompilerServices;

class Program
{
    private static readonly byte[] BytesArray = new byte[2] { 42, 43 };
    private static ReadOnlySpan<byte> BytesSpan => new byte[2] { 42, 43 };

    [MethodImpl(MethodImplOptions.NoInlining)]
    private static byte GetValueArray(int i) => BytesArray[i];

    [MethodImpl(MethodImplOptions.NoInlining)]
    private static byte GetValueSpan(int i) => BytesSpan[i];

    private static void Main()
    {
        Console.WriteLine(GetValueArray(1));
        Console.WriteLine(GetValueSpan(1));
    }
}
```

For GetValueArray:
```
G_M23015_IG01:
       56                   push     rsi
       4883EC20             sub      rsp, 32
       8BF1                 mov      esi, ecx
                                                ;; bbWeight=1    PerfScore 1.50
G_M23015_IG02:
       48B9905E2C50FD7F0000 mov      rcx, 0x7FFD502C5E90
       BA01000000           mov      edx, 1
       E8756EB85F           call     CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE
       48B8482C84D065010000 mov      rax, 0x165D0842C48
       488B00               mov      rax, gword ptr [rax]
       3B7008               cmp      esi, dword ptr [rax+8]
       730E                 jae      SHORT G_M23015_IG04
       4863D6               movsxd   rdx, esi
       0FB6441010           movzx    rax, byte  ptr [rax+rdx+16]
                                                ;; bbWeight=1    PerfScore 9.00
G_M23015_IG03:
       4883C420             add      rsp, 32
       5E                   pop      rsi
       C3                   ret
                                                ;; bbWeight=1    PerfScore 1.75
G_M23015_IG04:
       E8D0D5C95F           call     CORINFO_HELP_RNGCHKFAIL
       CC                   int3
```
and for GetValueSpan:
```
G_M49046_IG01:
       4883EC28             sub      rsp, 40
                                                ;; bbWeight=1    PerfScore 0.25
G_M49046_IG02:
       83F902               cmp      ecx, 2
       7316                 jae      SHORT G_M49046_IG04
       4863C1               movsxd   rax, ecx
       48BAAC285EC065010000 mov      rdx, 0x165C05E28AC
       0FB60410             movzx    rax, byte  ptr [rax+rdx]
                                                ;; bbWeight=1    PerfScore 3.75
G_M49046_IG03:
       4883C428             add      rsp, 40
       C3                   ret
                                                ;; bbWeight=1    PerfScore 1.25
G_M49046_IG04:
       E89C2AC95F           call     CORINFO_HELP_RNGCHKFAIL
       CC                   int3
```

It's even better when the JIT can see the value being indexed, as it can use the known span-length to eliminate the bounds checking, e.g. with this variation:
```C#
using System;
using System.Runtime.CompilerServices;

class Program
{
    private static readonly byte[] BytesArray = new byte[2] { 42, 43 };
    private static ReadOnlySpan<byte> BytesSpan => new byte[2] { 42, 43 };

    [MethodImpl(MethodImplOptions.NoInlining)]
    private static byte GetValueArray() => BytesArray[0];

    [MethodImpl(MethodImplOptions.NoInlining)]
    private static byte GetValueSpan() => BytesSpan[0];

    private static void Main()
    {
        Console.WriteLine(GetValueArray());
        Console.WriteLine(GetValueSpan());
    }
}
```

we got for GetValueArray:
```
G_M38228_IG01:
       4883EC28             sub      rsp, 40
                                                ;; bbWeight=1    PerfScore 0.25
G_M38228_IG02:
       48B9905E2D50FD7F0000 mov      rcx, 0x7FFD502D5E90
       BA01000000           mov      edx, 1
       E8786EB75F           call     CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE
       48B8482C001041020000 mov      rax, 0x24110002C48
       488B00               mov      rax, gword ptr [rax]
       83780800             cmp      dword ptr [rax+8], 0
       7609                 jbe      SHORT G_M38228_IG04
       0FB64010             movzx    rax, byte  ptr [rax+16]
                                                ;; bbWeight=1    PerfScore 8.75
G_M38228_IG03:
       4883C428             add      rsp, 40
       C3                   ret
                                                ;; bbWeight=1    PerfScore 1.25
G_M38228_IG04:
       E8D7D5C85F           call     CORINFO_HELP_RNGCHKFAIL
       CC                   int3
```

and for GetValueSpan:
```
G_M35269_IG01:
                                                ;; bbWeight=1    PerfScore 0.00
G_M35269_IG02:
       48B89828E06F41020000 mov      rax, 0x2416FE02898
       0FB600               movzx    rax, byte  ptr [rax]
                                                ;; bbWeight=1    PerfScore 2.25
G_M35269_IG03:
       C3                   ret
```

So, yes, there is upside.

Reply:
Thank you. 🎉